pytorch_lightning/accelerators/accelerator.py

```diff
@@ -26,8 +27,12 @@
 from pytorch_lightning.plugins.training_type import TrainingTypePlugin
 from pytorch_lightning.plugins.training_type.horovod import HorovodPlugin
 from pytorch_lightning.utilities.apply_func import move_data_to_device
+from pytorch_lightning.utilities.distributed import all_gather_ddp_if_available
 from pytorch_lightning.utilities.enums import AMPType, LightningEnum
 
+if TYPE_CHECKING:
+    from pytorch_lightning.trainer.trainer import Trainer
```

Comment:
this makes for some reason fail docs, so let's drop this typing for now...
(at least it is my observation from  #5919)